src/lazy/expanded/sequence.rs

```diff
@@ -202,6 +207,50 @@ impl<'top, D: Decoder> Iterator for ExpandedListIterator<'top, D> {
     }
 }
 
+/// Like an [`ExpandedListIterator`], but will yield macro invocations before also yielding
+/// that macro's expansion.
+#[derive(Debug)]
+pub struct ListValueExprIterator<'top, D: Decoder> {
```

Comment:
🪧 This section introduces iterators that return both the expressions _and_ the values they expand to.